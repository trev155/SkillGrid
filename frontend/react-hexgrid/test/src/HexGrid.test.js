import React from 'react';
import renderer from 'react-test-renderer';

import HexGrid from '../../src/HexGrid';

test('HexGrid should render correctly with default props', () => {
  const tree = renderer.create(
    <HexGrid>
      <div>child</div>
    </HexGrid>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('HexGrid should render correctly with custom props', () => {
  const tree = renderer.create(
    <HexGrid
      width={888}
      height={666}
      viewBox={'-150 -150 1100 1100'}
    >
      <div>child</div>
    </HexGrid>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('HexGrid should accept percentage in width and height', () => {
  const tree = renderer.create(
    <HexGrid
      width={'100%'}
      height={'50%'}
    >
      <div>child</div>
    </HexGrid>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
