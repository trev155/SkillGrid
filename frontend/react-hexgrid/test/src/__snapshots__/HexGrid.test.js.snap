// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HexGrid should accept percentage in width and height 1`] = `
<svg
  className="grid"
  height="50%"
  version="1.1"
  viewBox="-50 -50 100 100"
  width="100%"
  xmlns="http://www.w3.org/2000/svg"
>
  <div>
    child
  </div>
</svg>
`;

exports[`HexGrid should render correctly with custom props 1`] = `
<svg
  className="grid"
  height={666}
  version="1.1"
  viewBox="-150 -150 1100 1100"
  width={888}
  xmlns="http://www.w3.org/2000/svg"
>
  <div>
    child
  </div>
</svg>
`;

exports[`HexGrid should render correctly with default props 1`] = `
<svg
  className="grid"
  height={600}
  version="1.1"
  viewBox="-50 -50 100 100"
  width={800}
  xmlns="http://www.w3.org/2000/svg"
>
  <div>
    child
  </div>
</svg>
`;
